java/code/src/com/redhat/rhn/taskomatic/TaskoQuartzHelper.java

```diff
@@ -179,6 +182,29 @@ public static Date rescheduleJob(TaskoSchedule schedule, Instant startAtDate, lo
         return date;
     }
 
+    /**
+     * Reschedules a Job that is currently being executed based on the provided execution context.
+     * This method is used to retry a Job after a specified interval.
+     * @param context    the context of the current execution of the Job
+     * @param retryCount the current number of retries for this Job
+     * @param interval   the interval in seconds for triggering the Job again
+     * @throws SchedulerException if there is an issue with the scheduler while rescheduling the Job
+     */
+    public static void rescheduleJob(JobExecutionContext context, int retryCount, int interval)
+            throws SchedulerException {
+        Trigger oldTrigger = context.getTrigger();
+        oldTrigger.getJobDataMap().put("retryCount", retryCount + 1);
```

Comment:
Not easy to understand why you change the retryCount of the "oldTrigger".
I saw only later, that you create the new trigger by getting the Job data again.
Maybe it would be easier to understand when you spend a local variable for jobData, set the new retryCount there and provide this variable to "usingJobData()" function.

Reply:
I agree with Michael on this one.